src/build/__main__.py

```diff
@@ -57,54 +61,43 @@ def _format_dep_chain(dep_chain):  # type: (Sequence[str]) -> str
     return ' -> '.join(dep.partition(';')[0].strip() for dep in dep_chain)
 
 
-def _build_in_isolated_env(builder, outdir, distributions, config_settings):
-    # type: (ProjectBuilder, str, List[str], ConfigSettingsType) -> None
-    for distribution in distributions:
-        with IsolatedEnvBuilder() as env:
-            builder.python_executable = env.executable
-            builder.scripts_dir = env.scripts_dir
-            # first install the build dependencies
-            env.install(builder.build_system_requires)
-            # then get the extra required dependencies from the backend (which was installed in the call above :P)
-            env.install(builder.get_requires_for_build(distribution))
-            builder.build(distribution, outdir, config_settings)
-
-
-def _build_in_current_env(builder, outdir, distributions, config_settings, skip_dependency_check=False):
-    # type: (ProjectBuilder, str, List[str], ConfigSettingsType, bool) -> None
-    for dist in distributions:
-        if not skip_dependency_check:
-            missing = builder.check_dependencies(dist)
-            if missing:
-                _error(
-                    'Missing dependencies:'
-                    + ''.join('\n\t' + dep for deps in missing for dep in (deps[0], _format_dep_chain(deps[1:])) if dep)
-                )
+def _build_in_isolated_env(builder, outdir, distribution, config_settings):
+    # type: (ProjectBuilder, str, str, Optional[ConfigSettingsType]) -> str
+    with IsolatedEnvBuilder() as env:
+        builder.python_executable = env.executable
+        builder.scripts_dir = env.scripts_dir
+        # first install the build dependencies
+        env.install(builder.build_system_requires)
+        # then get the extra required dependencies from the backend (which was installed in the call above :P)
+        env.install(builder.get_requires_for_build(distribution))
+        return builder.build(distribution, outdir, config_settings or {})
 
-        builder.build(dist, outdir, config_settings)
 
+def _build_in_current_env(builder, outdir, distribution, config_settings, skip_dependency_check=False):
+    # type: (ProjectBuilder, str, str, Optional[ConfigSettingsType], bool) -> str
+    if not skip_dependency_check:
+        missing = builder.check_dependencies(distribution)
+        if missing:
+            _error(
+                'Missing dependencies:'
+                + ''.join('\n\t' + dep for deps in missing for dep in (deps[0], _format_dep_chain(deps[1:])) if dep)
```

Comment:
Assign this string into a variable so that it fits in one line. Will make it easier to read.

Reply:
This was the old code, which I did not touch, but we can take the opportunity to improve :blush: